### PR TITLE
Create ambient docs test job

### PR DESCRIPTION
**Pull request by kfaseela**
Needed for https://github.com/istio/istio.io/pull/15210

### CODE REVIEW

File: prow/config/jobs/istio.io.yaml [hunk 1]
```diff
@@ -31,6 +31,11 @@ jobs:
     command: [entrypoint, prow/integ-suite-kind.sh, doc.test.profile-minimal]
     requirements: [kind]
     regex: ^(go.mod$|tests/|pkg/test/|prow/|content/en/boilerplates/snips/|content/en/docs/.*(test\.sh|/snips\.sh)$)
+  
+  - name: doc.test.profile-ambient
+    command: [entrypoint, prow/integ-suite-kind.sh, doc.test.profile-ambient]
+    requirements: [kind]
```
Comment (dhawton) [inline prow/config/jobs/istio.io.yaml:37]:
```suggestion
    requirements: [kind]
    modifiers: [presubmit_optional]
```

**Review by dhawton — COMMENTED** [review]
Might be worth making optional until we get the docs PR merged and validate it is working as expected, otherwise it'll stop docs PRs from merging until all issues are resolved.

**Review by dhawton — APPROVED** [review]
LGTM

**Comment by kfaseela** [comment]
Oops.. I need to run make gen again..

**Pull request merged**